packages/connect/src/protocol/async-iterable-story.spec.ts

```diff
@@ -265,11 +262,16 @@ describe("full story", function () {
         .then(() => fail("send was unexpectedly resolved."))
         .catch((e) => expect(e).toBe("READER_ERROR"));
 
-      // The reader's internal writer is closed so any future reads will be rejected.
+      // The reader's internal writer is closed so any future reads should result in the
+      // done.
       readerIt[Symbol.asyncIterator]()
         .next()
-        .then(() => fail("reads were unexpectedly resolved"))
-        .catch((e) => expect(e).toBe("READER_ERROR"));
+        .then((result) =>
+          expect(result).toEqual({ done: true, value: undefined })
+        )
+        .catch(() =>
+          fail("expected successful done result but unexpectedly rejected")
+        );
```

Comment:
Changed the behavior for next to return `{done: true}` when the writer is closed either due to an error or with call of `close`. This is what I think should be the behavior of Iterables outlined [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#the_iterator_protocol), based on these two paragraphs:
>If an iterator returns a result with done: true, any subsequent calls to next() are expected to return done: true as well, although this is not enforced on the language level.

>throw(exception) Optional
A function that accepts zero or one argument and returns an object conforming to the IteratorResult interface, typically with done equal to true. Calling this method tells the iterator that the caller detects an error condition, and exception is typically an [Error](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error) instance.